### DISCUSSION TITLE
convexHull with JAX not working

### DISCUSSION THREAD

**Discussion by jacquesbarman27** [jax-ml/jax#15381] [Q&A]
Hello, 

I am trying to use JAX for an optimization problem. In my problem, I need to compute the volume and surface area of a geometry given by its vertices. To do this, I use the function convexHull from "from scipy.spatial import ConvexHull". Unfortunately, it seems like JAX can't follow the scipy object and it gives me an error. The issue seems to be with the grad function. Is there a way to use convexHull with JAX? Any ideas on how to fix this? 

Thank you!

![image](https://user-images.githubusercontent.com/78436570/229667222-7befe1bd-0bdd-48ea-a209-2d9f6532cbe8.png)
![image](https://user-images.githubusercontent.com/78436570/229667563-bdd63071-211d-4f62-b399-e5af24018e20.png)
![image](https://user-images.githubusercontent.com/78436570/229667673-9a50c804-f750-4eee-9e49-0b4bfbe43726.png)

**Comment by jakevdp** [comment]
Hi - thanks for the question! JAX transformations do not work with arbitrary Python code, and in particular will not work with `scipy`  code like `scipy.spatial.qhull`. If you want to use this kind of functionality with JAX, you'll need to find a convex hull implementation written for JAX.

**Reply by jacquesbarman27** [reply]
I see, thank you for the answer! Are you aware of any convex hull implementation written for JAX, or if it's in the plan for a future release? If not, is there a way around it to be able to implement the convexHull from scipy with JAX for the moment?

Thank you!

**Reply by jakevdp** [reply]
No, unfortunately I don't know of any convexhull implementations in JAX, and I haven't heard of anyone working on that.
